### PR TITLE
removed enable-hostpath-provisioner entry

### PR DESCRIPTION
**Pull request by deepakm-ntnx**
reason: it was carry forward from docker provider and is not needed for nutanix provider kcp and kcpt

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**How Has This Been Tested?**:

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and test output_

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by deepakm-ntnx** [comment]
local testing results with quick-start test 
<pre>
make test-e2e-calico LABEL_FILTERS=quickstart  
</pre>
Output
<pre>
When following the Cluster API quick-start Should create a workload cluster [quickstart, capx-feature-test]
/Users/deepak.muley/go/pkg/mod/sigs.k8s.io/cluster-api/test@v1.3.5/e2e/quick_start.go:78
  STEP: Creating a namespace for hosting the "quick-start" test spec @ 01/19/24 15:29:10.682
  INFO: Creating namespace quick-start-rudnef
  INFO: Creating event watcher for namespace "quick-start-rudnef"
  STEP: Creating a workload cluster @ 01/19/24 15:29:10.709
  INFO: Creating the workload cluster with name "quick-start-z5lu20" using the "(default)" template (Kubernetes v1.27.9, 1 control-plane machines, 1 worker machines)
  INFO: Getting the cluster template yaml
  INFO: clusterctl config cluster quick-start-z5lu20 --infrastructure (default) --kubernetes-version v1.27.9 --control-plane-machine-count 1 --worker-machine-count 1 --flavor (default)
  INFO: Applying the cluster template yaml to the cluster
configmap/quick-start-z5lu20-pc-trusted-ca-bundle created
configmap/nutanix-ccm created
configmap/cni-quick-start-z5lu20-crs-cni created
secret/quick-start-z5lu20 created
secret/nutanix-ccm-secret created
clusterresourceset.addons.cluster.x-k8s.io/nutanix-ccm-crs created
clusterresourceset.addons.cluster.x-k8s.io/quick-start-z5lu20-crs-cni created
kubeadmconfigtemplate.bootstrap.cluster.x-k8s.io/quick-start-z5lu20-kcfg-0 created
cluster.cluster.x-k8s.io/quick-start-z5lu20 created
machinedeployment.cluster.x-k8s.io/quick-start-z5lu20-wmd created
machinehealthcheck.cluster.x-k8s.io/quick-start-z5lu20-mhc created
kubeadmcontrolplane.controlplane.cluster.x-k8s.io/quick-start-z5lu20-kcp created
nutanixcluster.infrastructure.cluster.x-k8s.io/quick-start-z5lu20 created
nutanixmachinetemplate.infrastructure.cluster.x-k8s.io/quick-start-z5lu20-mt-0 created

  INFO: Waiting for the cluster infrastructure to be provisioned
  STEP: Waiting for cluster to enter the provisioned phase @ 01/19/24 15:29:13.31
  INFO: Waiting for control plane to be initialized
  INFO: Waiting for the first control plane machine managed by quick-start-rudnef/quick-start-z5lu20-kcp to be provisioned
  STEP: Waiting for one control plane node to exist @ 01/19/24 15:29:23.348
  INFO: Waiting for control plane to be ready
  INFO: Waiting for control plane quick-start-rudnef/quick-start-z5lu20-kcp to be ready (implies underlying nodes to be ready as well)
  STEP: Waiting for the control plane to be ready @ 01/19/24 15:30:23.432
  STEP: Checking all the control plane machines are in the expected failure domains @ 01/19/24 15:30:23.44
  INFO: Waiting for the machine deployments to be provisioned
  STEP: Waiting for the workload nodes to exist @ 01/19/24 15:30:23.471
  STEP: Checking all the machines controlled by quick-start-z5lu20-wmd are in the "<None>" failure domain @ 01/19/24 15:30:53.519
  INFO: Waiting for the machine pools to be provisioned
  STEP: PASSED! @ 01/19/24 15:30:53.566
</pre>

File: test/e2e/data/infrastructure-nutanix/v1.2.4/base/cluster-with-kcp.yaml
Comment (thunderboltsid) [inline test/e2e/data/infrastructure-nutanix/v1.2.4/base/cluster-with-kcp.yaml:1]: Please remove this change from the template on this file. A versioned template should be reflective of the state of affairs at the time of the release. Let's not pollute the versioned templates when changing it for main.

File: test/e2e/data/infrastructure-nutanix/v1.3.0/base/cluster-with-kcp.yaml
Comment (thunderboltsid) [inline test/e2e/data/infrastructure-nutanix/v1.3.0/base/cluster-with-kcp.yaml:1]: Please remove this change from the template on this file. A versioned template should be reflective of the state of affairs at the time of the release. Let's not pollute the versioned templates when changing it for main.

**Review by thunderboltsid — CHANGES_REQUESTED** [review]
LGTM except for modification of versioned templates we use for tests.

**Review by thunderboltsid — APPROVED** [review]

**Review by adiantum — APPROVED** [review]
/lgtm
/approve

**Pull request merged**